hxmath/math/DualQuaternion.hx

```diff
@@ -0,0 +1,644 @@
+     * @param t             The Vector4 ( translation )
+     * @return              The DualQuaternion
+     */
+    public static inline function fromAxisRadian( theta:Float
+                                                , axis: Vector3
+                                                , vec4: Vector4 ):DualQuaternion 
+    {
+        var cosHalfAngle = Math.cos(theta / 2.0);
+        var sinHalfAngle = Math.sin(theta / 2.0);
+        var real = new Quaternion(cosHalfAngle,sinHalfAngle * axis.x,sinHalfAngle * axis.y,
+        sinHalfAngle * axis.z);
+        return create(real,vec4);
+    }
+    
+    /**
+     * Extract the Vector4 part of a DualQuaternion
+     * 
+     * @return          The translation Vector
+     */
+    public inline function getTranslation(): Vector4 
+    {
+        var q = this.dual * (~this.real) * 2.0;
+        return new Vector4(q.x, q.y, q.z, 1.0);
+    }
+    
+    /**
+     * Extract the Real Quaternion the 3D angle as a copy
+     * Particularly useful as it clarifies that the Rotation
+     * 
+     * @return         The rotational Quaternion
+     */
+    public inline function getRotationQuat(): Quaternion 
+    {
+        return this.real.clone();
+    }
+    
+    /**
+     * Construct a DualQuaternion from an array.
+     * 
+     * @param rawData   The input array.
+     * @return          The constructed structure.
+     */
+    @:from
+    public static inline function fromArray(rawData:Array<Float>):DualQuaternion
+    {
+        if (rawData.length != DualQuaternion.elementCount)
+        {
+            throw "Invalid rawData.";
+        }
+        var real = new Quaternion(rawData[0], rawData[1], rawData[2], rawData[3]);
+        var dual = new Quaternion(rawData[4], rawData[5], rawData[6], rawData[7]);
+        return new DualQuaternion(real, dual);
+    }
+    
+    /**
+     * Creates a Scalar DualQuaternion
+     * 
+     * @param s         The scale value
+     * @return          The Scalar DualQuaternion
+     */
+    public static inline function scalar(s: Float):DualQuaternion
+    {
+        return new DualQuaternion(Quaternion.scalarReal( s ), Quaternion.zero);
+    }
+    
+    /**
+     * Clone.
+     * 
+     * @return          The cloned object.
+     */
+    public inline function clone():DualQuaternion
+    {
+        return new DualQuaternion( this.real.clone(), this.dual.clone() );
+    }
+    
+    /**
+     * dot product
+     *
+     * @param a         one DualQuaternion
+     * @param b         another DualQuaternion
+     * @return          the dot product.
+     */
+    public static inline function dot(a:DualQuaternion, b:DualQuaternion):Float
+    {
+        return Quaternion.dot(a.real.clone(), b.real.clone());
+    }
+    
+    /**
+     * Create a complex conjugate copy of a dualquaternion
+     * 
+     * @param a
+     * @return       ~a
+     */
+    @:op(~A)
+    public static inline function conjugate(a: DualQuaternion):DualQuaternion
```

Comment:
See comment below. Should either be skipped, or the documentation should make it clear this is the complex conjugate. I could also see this being implemented as the complex + dual conjugate. This is the way transformation by a dual quaternion is implemented: q * v * complexConjugate(dualConjugate(q)).

Reply:
I think you have lost me here.